### PR TITLE
Trip Backend

### PR DESCRIPTION
**Pull request by usamazansari**

### CODE REVIEW

File: apps/api/src/main.ts [hunk 1]
```diff
@@ -1,22 +1,19 @@
...
- * This is not a production server yet!
- * This is only a minimal backend to get started.
- */
-
+import { LoggerService } from '@itinerary-assistant/api/core';
 import { Logger } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
 import { NestFactory } from '@nestjs/core';
 
 import { AppModule } from './app/app.module';
```
Comment (usama251993) [inline apps/api/src/main.ts:6]: Organize the imports

File: libs/api/database/neo4j/src/lib/neo4j.module.ts [hunk 2]
```diff
@@ -40,16 +38,20 @@ export class Neo4jModule {
...
               return connection;
             } catch (error: unknown) {
-              console.log(
-                `Error in connection - ${(<ConnectionErrorType>error).code}`
+              Logger.error(
+                `Error in connection - ${(<ConnectionErrorType>error).code}`,
+                'Neo4jModule'
               );
-              console.log(`Could not connect to Neo4j`);
+              Logger.error(`Could not connect to Neo4j`, 'Neo4jModule');
```
Comment (usama251993) [inline libs/api/database/neo4j/src/lib/neo4j.module.ts:54]: Use Custom Logger Service

File: libs/api/database/neo4j/src/lib/services/query-repository/query-repository.service.ts [hunk 3]
```diff
@@ -10,7 +15,7 @@ class QueryRepositoryService implements OnApplicationShutdown {
   ) {}
 
   onApplicationShutdown(): void {
-    console.log('Neo4j Shutdown');
+    Logger.log(`${this.constructor.name} ❌ Shutdown`, 'Neo4jModule');
```
Comment (usama251993) [inline libs/api/database/neo4j/src/lib/services/query-repository/query-repository.service.ts:18]: Use custom logger service

File: libs/api/database/neo4j/src/lib/utils/extractor/extractor.utility.ts [hunk 4]
```diff
@@ -1,12 +1,49 @@
...
+ * extractEntity<T>(result) = {
+ *   id: '',
+ *   ...props
+ * }
+ * ```
+ *
+ * @param r Neo4jOutput<T>
+ * @returns T[]
+ */
 const extractEntity = <T = unknown>(r: Neo4jOutput<T>) =>
```
Comment (usama251993) [inline libs/api/database/neo4j/src/lib/utils/extractor/extractor.utility.ts:32]: Rewrite as `<T extends BaseModel>` instead of `<T = unknown>`

**Review by usama251993 — CHANGES_REQUESTED** [review]
Resolve the comments

File: libs/api/database/neo4j/src/lib/services/query-repository/query-repository.service.ts (hunk 3 above)
Comment (usamazansari) [reply libs/api/database/neo4j/src/lib/services/query-repository/query-repository.service.ts:18]: Cannot use logger service due to circular dependencies despite using `forwardRef()`

**Review by usama251993 — APPROVED** [review]
OK

**Pull request merged**